src/client/actions/test/edit/articleActions.test.ts

```diff
@@ -111,52 +134,69 @@ describe("articleActions", () => {
+      dispatch.mockClear()
+      dispatchedChangeArticleData(dispatch, getState)
+
+      expect(dispatch).toBeCalledWith({
+        payload: { data: { title: "New Title" } },
+        type: "CHANGE_ARTICLE",
+      })
     })
 
-    it("calls debounced #updateArticle with new attrs", done => {
+    it("calls debounced #updateArticle with new attrs", () => {
+      global.Date = jest.fn(() => ({
+        toISOString: () => "2019-03-19T20:33:06.821Z",
+      })) as any
       onChangeArticle("title", "New Title")(dispatch, getState)
-      dispatch.mock.calls[0][0](dispatch, getState)
-      setTimeout(() => {
-        expect(dispatch.mock.calls[2][0].type).toBe("UPDATE_ARTICLE")
-        expect(dispatch.mock.calls[2][0].key).toBe("userCurrentlyEditing")
-        expect(dispatch.mock.calls[2][0].payload.article).toBe(article.id)
-        done()
-      }, 550)
+
+      expect(dispatch).toBeCalledWith({
+        type: "UPDATE_ARTICLE",
+        key: "userCurrentlyEditing",
+        payload: {
+          article: "594a7e2254c37f00177c0ea9",
+          channel: {
+            type: "editorial",
+          },
+          timestamp: "2019-03-19T20:33:06.821Z",
+        },
+      })
     })
 
     it("does not call #saveArticle if published", () => {
       onChangeArticle("title", "New Title")(dispatch, getState)
-      dispatch.mock.calls[0][0](dispatch, getState)
-      expect(dispatch.mock.calls.length).toBe(2)
+      const dispatchedChangeArticleData = dispatch.mock.calls[0][0]
+      dispatch.mockClear()
+      dispatchedChangeArticleData(dispatch, getState)
+
+      expect(dispatch).toBeCalledWith({
+        payload: { data: { title: "New Title" } },
+        type: "CHANGE_ARTICLE",
+      })
+      expect(Backbone.sync).not.toBeCalled()
     })
 
-    it("calls debounced #saveArticle if draft", done => {
+    it("calls debounced #saveArticle if draft", async () => {
       article.published = false
-      onChangeArticle("title", "N")(dispatch, getState)
-      dispatch.mock.calls[0][0](dispatch, getState)
-      onChangeArticle("title", "Ne")(dispatch, getState)
-      dispatch.mock.calls[2][0](dispatch, getState)
-      onChangeArticle("title", "New")(dispatch, getState)
-      dispatch.mock.calls[4][0](dispatch, getState)
+      onChangeArticle("title", "New title")(dispatch, getState)
+      const dispatchedChangeArticleData = dispatch.mock.calls[0][0]
+      dispatchedChangeArticleData(dispatch, getState)
+      const dispatchedSaveArticle = dispatch.mock.calls[2][0]
+      dispatchedSaveArticle(dispatch, getState)
 
-      setTimeout(() => {
-        expect(dispatch.mock.calls.length).toBe(8)
-        done()
-      }, 550)
+      expect(Backbone.sync.mock.calls[0][0]).toBe("update")
     })
   })
 
   describe("#publishArticle", () => {
     let getState
     let dispatch
-    let setArticleSpy = jest.spyOn(Article.prototype, "set")
+    const setArticleSpy = jest.spyOn(Article.prototype, "set")
```

Comment:
What's the different in using `spyOn()` versus `fn()` here?

Reply:
`spyOn` lets you track calls to the function without overwriting/effecting its implementation

Reply:
Gotcha